src/main/java/org/prebid/server/bidder/gotthamads/GothamAdsBidder.java

```diff
@@ -0,0 +1,145 @@
+package org.prebid.server.bidder.gotthamads;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.gothamads.GothamAdsImpExt;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class GothamAdsBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, GothamAdsImpExt>> TYPE_REFERENCE = new TypeReference<>() {
+    };
+    private static final String ACCOUNT_ID_MACRO = "{{AccountId}}";
+    private static final String X_OPENRTB_VERSION = "2.5";
+    private static final Map<Integer, BidType> ALLOWED_IMP_TYPES = Map.of(
+            1, BidType.banner,
+            2, BidType.video,
+            4, BidType.xNative
+    );
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public GothamAdsBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final GothamAdsImpExt impExt;
+        final Imp imp = request.getImp().get(0);
+        try {
+            impExt = parseImpExt(imp);
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+
+        final BidRequest bidRequest = request.toBuilder()
+                .imp(List.of(cleanUpExt(imp)))
+                .build();
+
+        final HttpRequest<BidRequest> httpRequest = HttpRequest.<BidRequest>builder()
```

Comment:
Please, add information of which impids current http request covers, it is needed for seatnonbid infrastructure and can be provided by adding:
```java
.impIds(BidderUtil.impIds(bidRequest))
```